src/BaseWrapper.sol

```diff
@@ -85,4 +89,30 @@ abstract contract BaseWrapper is IERC3156PPFlashLender {
     function _repayTo() internal view virtual returns (address) {
         return address(this);
     }
+
+    /// @dev Measure and record gas used in flash loans
+    function setExpectedGas(IERC20 asset) external returns (uint256 gasUsed){
+        uint256 gasLeftBefore = gasleft();
+        this.flashLoan(address(this), asset, 10 ** asset.decimals(), "", this.gasCallback);
+
+        gasUsed = gasLeftBefore - gasleft();
+        expectedGas = gasUsed;
+        emit GasUsed(gasUsed);
+    }
+
+    /// @dev Callback function used to measure gas used in flash loans
+    function gasCallback(
+        address,
+        address repayTo,
+        IERC20 asset,
+        uint256 amount,
+        uint256 fee,
+        bytes memory
+    )
+        external
+        returns (bytes memory)
+    {
+        if (repayTo != address(0)) _transferAssets(asset, amount + fee, repayTo);
+        return "";
+    }
```

Comment:
I see what it does, but what exactly are you trying to achieve? is it for reporting purposes?

Reply:
My thought here is that some borrowers might want to know the expected gas on-chain so that they can factor that cost into the fees.

Reply:
But on chain wouldn't make sense, and for doing it off chain, they'd be better off calculating the whole tx and looking at the end result than just at this.
It doesn't hurt to have it, but I'm not sure anyone's gonna use it.

Reply:
I think it would be useful, but also think that less is more, let's remove it.